### DISCUSSION TITLE
Support of 8 channel 12 bit AD converter MCP3208

### DISCUSSION THREAD

**Discussion by sternegugger** [arendst/Tasmota#13392] [Ideas]
Hello tasmota developers,

for my project I need to read the analog outputs from 8 temperature sensors. The ESP8266 has only one analog input.
So I think the [MCP3208](https://www.reichelt.de/index.html?ACTION=7&LA=3&OPEN=0&INDEX=0&FILENAME=A200%2FMCP3204_MCP3208_MIC.pdf) is a good extension. It has 8 analog channels (12 bit resolution) and can be connected via SPI bus to the ESP.

Before I have learn how to extend the tasmota code, I would like to ask the following questions:

- Is the MCP3208 already supported? If yes, which compile option do I have to use?
- Is any other 8 channel A/D converter supported? I found only the [ADS111x](https://tasmota.github.io/docs/Supported-Peripherals/), which supports maximum only 4 analog channels.
- Is it possible to use multiple ADS111x with tasmota at one ESP8266 to get more analog inputs?

- In case all answers above are "no", where can I find a manual with coding rules to extend tasmota to support an other SPI device (MCP3208)?

Thank you. 
Tasmota is great and helped me a lot with my smart home projects.

**Comment by barbudor** [comment]
You can have up to 4 x ADS1115 which makes 16 analog inputs
You can also use ESP32 that has up to 8 native ADC inputs in Tasmota

If you want to build you own driver, start reading https://tasmota.github.io/docs/For-Developers/
and get inspiration from other similar drivers

**Reply by sfromis** [reply]
Submitting a PR for MCP3208 would be welcome. Tasmota generally "prefer" I2C for sensors, SPI might be "less easy" to create a driver for.

**Reply by sternegugger** [reply]
> You can also use ESP32 that has up to 8 native ADC inputs in Tasmota

Thank you barbudor for your fast reply.
Using the ESP32 seems to be the easiest and cheapest way to get enough analog inputs. I have already ordered some.

**Reply by sternegugger** [reply]
> Submitting a PR for MCP3208 would be welcome. Tasmota generally "prefer" I2C for sensors, SPI might be "less easy" to create a driver for.

I am not sure what you mean with "PR". May by the data sheet of the  MCP3208 could help:
https://cdn-reichelt.de/documents/datenblatt/A200/MCP3204_MCP3208_MIC.pdf

**Comment by barbudor** [comment]
PR (pull request) means that you do the implementation of the driver and you submit your code for inclusion in Tasmota
